Graphene SGX not building with DEBUGOPT=1

<!-- Thanks for reporting a bug.

Please provide as much information as possible.
Paste any logs using three backticks (```).

IMPORTANT: please do not post general questions here;
if you have questions, please use mailing list: support@graphene-project.io
-->

## Description of the problem

To collect the perf record for workloads I was trying to following the instructions in the Graphene docs for SGX profiling and there it mentions about building Graphene with SGX=1 and DEBUGOPT=1 flag but when I tried that on the latest Graphene I am getting build errors. Please find attached the build log with the error.
```
In file included from ../../../../common/include/api.h:440,
                 from ../../../include/pal_internal.h:14,
                 from elf-x86_64.h:11,
                 from sgx_profile.c:19:
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/features.h:424,
                 from /usr/include/errno.h:25,
                 from sgx_profile.c:14:
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
../../../../common/include/api_fortified.h:34:25: error: expected declaration specifiers or ‘...’ b
   34 | #define __api_bos0(ptr) __builtin_object_size(ptr, 0)
      |                         ^~~~~~~~~~~~~~~~~~~~~
Makefile:116: recipe for target 'sgx_profile.o' failed
```
When I try to build the same for Graphene native, the build is successful. 


## Steps to reproduce
<!-- NOTE: please specify the exact commit ID on which you reproduced the issue

### PLEASE ENSURE THAT THE ISSUE REPRODUCES ON THE CURRENT MASTER BRANCH ###

-->

make SGX=1 DEBUGOPT=1 ISGX_DRIVER_PATH=""
[SGX_Debugopts.log](https://github.com/oscarlab/graphene/files/6667221/SGX_Debugopts.log)


## Expected results

## Actual results

<!-- ## Additional information -->
<!-- if applicable, uncomment and fill this section -->

--- a/Pal/src/host/Linux-SGX/sgx_profile.c
+++ b/Pal/src/host/Linux-SGX/sgx_profile.c
@@ -10,6 +10,8 @@
 
 #ifdef DEBUG
 
+#define USE_STDLIB
+
 #include <assert.h>
 #include <errno.h>
 #include <linux/limits.h>
@@ -23,7 +25,6 @@
 #include "sgx_log.h"
 #include "sgx_tls.h"
 #include "spinlock.h"
-#include "string.h"
 
 // FIXME: this is glibc realpath, declared here because the headers will conflict with PAL
 char* realpath(const char* path, char* resolved_path);

--- a/common/include/api_fortified.h
+++ b/common/include/api_fortified.h
@@ -13,9 +13,13 @@
  * - https://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html
  * - glibc sources: string_fortified.h, stdio2.h
  *
- * This file will conflict with regular glibc headers (e.g. stdio.h, string.h). You should not use
- * these headers in no-stdlib context, but if necessary, you can disable this file by defining
- * USE_STDLIB before including api.h:
+ * This file will conflict with regular libc headers (e.g. <stdio.h>, <string.h>). That should not
+ * be an issue because in no-stdlib context you should not use most libc headers, and in stdlib
+ * context you should not use "api.h".
+ *
+ * However, if for any reason you need to include both "api.h" and headers such as <stdio.h>
+ * alongside each other, you can disable this file (and prevent the conflict) by defining
+ * USE_STDLIB:
  *
  *     #include <stdio.h>
  *     #include <string.h>

